cartesi-rollups/development/migration.md

```diff
@@ -1,44 +1,148 @@
+- `RPC_URL`: The RPC endpoint to be used.
+- `MNEMONIC`: The mnemonic phrase for the Authority owner's wallet (other wallet options may be used).
+- `HISTORY_FACTORY_ADDRESS`: The address of a valid HistoryFactory instance.
+- `AUTHORITY_ADDRESS`: The address of the Authority instance used by the application.
+
+  :::note environment variables
+  A `HistoryFactory` is deployed at `0x1f158b5320BBf677FdA89F9a438df99BbE560A26` for all supported networks, including Ethereum, Optimism, Arbitrum, Base, and their respective Sepolia-based testnets.
+  :::
+
+#### 2. Instantiate a New _History_
+
+This is a two-step process. First calculate the address of the new History. After that, the new instance of History may be created.
+
+- To calculate the address of a new _History_ contract call the `calculateHistoryAddress(address,bytes32)(address)` function with the help of Foundry's Cast:
+
+  ```shell
+  cast call \
+  	--trace --verbose \
+  	$HISTORY_FACTORY_ADDRESS \
+  	"calculateHistoryAddress(address,bytes32)(address)" \
+  	$AUTHORITY_ADDRESS \
+  	$SALT \
+  	--rpc-url "$RPC_URL"
+  ```
+
+  If the command executes successfully, it will display the address of the new History contract. Store this address in the environment variable `NEW_HISTORY_ADDRESS` for later use.
+
+- Create a new instance of _History_ may be created by calling function `newHistory(address,bytes32)`:
+
+  ```shell
+  cast send \
+  --json \
+  --mnemonic "$MNEMONIC" \
+  $HISTORY_FACTORY_ADDRESS \
+  "newHistory(address,bytes32)(History)" \
+  $AUTHORITY_ADDRESS \
+  $SALT \
+  --rpc-url "$RPC_URL"
+  ```
+
+  The `cast send` command will fail if the `History` type is not recognized by Cast at the time of execution. In such cases, replace `History` with `address` as the return type for `newHistory()` and execute the command again.
+
+  Additionally, the `cast send` command may fail due to gas estimation issues. To circumvent this, provide gas constraints with the `--gas-limit` parameter (e.g., `--gas-limit 7000000`).
+
+#### 3. Replace the _History_
+
+Make sure the environment variables from the previous step are set, including `NEW_HISTORY_ADDRESS`, which should have the address of the new History.
+
+To replace the _History_ used by the _Authority_, run this command:
+
+```shell
+cast send \
+    --json \
+    --mnemonic "$MNEMONIC" \
+    "$AUTHORITY_ADDRESS" \
+    "setHistory(address)" \
+    "$NEW_HISTORY_ADDRESS" \
+    --rpc-url "$RPC_URL"
+```
+
+After replacing the _History_, update the `CARTESI_CONTRACTS_HISTORY_ADDRESS` in the application configuration with the new _History_ address. Then, upgrade the Cartesi Rollups Node as usual.
+
+When the Cartesi Rollups Node restarts, it will process all existing inputs, recalculate the epochs, and send the claims to the new _History_ based on the updated configuration.
+
+---
+
+## Migrating to Cartesi CLI
+
 Follow these simple steps to migrate your application from Sunodo CLI to Cartesi CLI.
```

Comment:
Should we add CLI versions as well to give more clarity? I was a little confused as to whether this is 1.5 migration or not.

Reply:
This is a migration guide for Sunodo CLI to Cartesi CLI so perhaps the heading needs an update.

Reply:
@masiedu4 How relevant is it for us to add this migration in 1.5 docs? I mean devs will by default use Cartesi CLI with Rollups 1.5.

Reply:
Great point. I think we can remove this migration guide at this point @riseandshaheen  🤔